saltstack/base/salt/prerequisites/user.sls

```diff
@@ -1,3 +1,9 @@
+{% if salt['environ.get']('OS').startswith("centos") %}
+centos_user_noshell:
+  cmd.run:
```

Comment:
https://docs.saltproject.io/en/latest/ref/states/all/salt.states.user.html
user.present.shell would be a nicer way to set this